jsonschema/src/keywords/format.rs

```diff
@@ -281,12 +281,28 @@ impl Validate for JSONPointerValidator {
         }
     }
 }
-format_validator!(RegexValidator, "regex");
+struct RegexValidator {
+    schema_path: JSONPointer,
+    config: Arc<CompilationOptions>,
```

Comment:
Maybe keeping the regex engine options will be enough, instead of storing the whole set of options? Thinking more about just storing the necessary minimum, rather than performance. However, I think as cloning `RegexEngine` is cheap, it could be also better performance wise during validation (though for a cost of a pointer indirection)